lib/variable/arithmetic.cpp

```diff
@@ -0,0 +1,122 @@
+// SPDX-License-Identifier: BSD-3-Clause
+// Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+#include "scipp/variable/arithmetic.h"
+#include "scipp/core/dtype.h"
+#include "scipp/core/eigen.h"
+#include "scipp/core/element/arithmetic.h"
+#include "scipp/core/spatial_transforms.h"
+#include "scipp/variable/astype.h"
+#include "scipp/variable/pow.h"
+#include "scipp/variable/transform.h"
+#include "scipp/variable/variable_factory.h"
+
+namespace scipp::variable {
+
+namespace {
+
+bool is_transform_with_translation(const Variable &var) {
+  return var.dtype() == dtype<Eigen::Affine3d> ||
+         var.dtype() == dtype<scipp::core::Translation>;
+}
+
+auto make_factor(const Variable &prototype, const double value) {
+  const auto unit = variableFactory().elem_unit(prototype) == units::none
+                        ? units::none
+                        : units::one;
+  return astype(makeVariable<double>(Values{value}, unit),
+                variableFactory().elem_dtype(prototype));
+}
+
+/// True if a and b are correlated, currently only if referencing same.
+bool correlated(const Variable &a, const Variable &b) {
+  return variableFactory().has_variances(a) &&
+         variableFactory().has_variances(b) && a.is_same(b);
```

Comment:
Would it be enough to use `a.is_same(b)`? The computation for values is equivalent and potentially faster.

Reply:
I used that initially but it causes trouble in some cases, e.g., with dtypes that do not support `*` (such as `vector`).

Reply:
Ok